dependencies.yaml

```diff
@@ -93,29 +113,41 @@ dependencies:
               - sysroot_linux-aarch64==2.17
   py_build:
     common:
-      - output_types: [conda, requirements]
+      - output_types: [conda, requirements, pyproject]
         packages:
           - cuda-python>=11.7.1,<12.0
           - scikit-build>=0.13.1
           - cython>=0.29,<0.30
-          - treelite=3.1.0
-      - output_types: conda
+          - &treelite treelite==3.1.0
+          - pylibraft==23.4.*
+          - rmm==23.4.*
+      - output_types: pyproject
         packages:
-          - cudf=23.04.*
-          - raft-dask=23.04.*
-          - pylibraft=23.04.*
+          - wheel
+          - setuptools
+          - &treelite_runtime treelite_runtime==3.1.0
   py_run:
     common:
-      - output_types: [conda, requirements]
+      - output_types: [conda, requirements, pyproject]
         packages:
+          - cudf==23.4.*
+          - cupy>=9.5.0,<12.0.0a0
           - dask>=2023.1.1
+          - dask-cuda==23.4.*
+          - dask-cudf==23.4.*
           - distributed>=2023.1.1
           - joblib>=0.11
-      - output_types: conda
+          - numba
+            # TODO: Are seaborn and scipy really hard dependencies, or should
```

Comment:
seaborn is a dependency for notebooks and some experimental plotting functions.

scipy is for CPU sparse matrices currently

Reply:
In the interest of not backing up CI any further I'm going to merge this PR as is and can clean up these TODOs in the next PR since there will be more follow-ups improving dependencies.yaml.